catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -204,6 +213,23 @@ QueryResponse doQuery(QueryRequest queryRequest, FederationStrategy strategy)
 
         LOGGER.debug("Calling strategy.federate()");
 
+        Query originalQuery = queryRequest.getQuery();
+
+        long timeout = originalQuery.getTimeoutMillis();
+
+        if (timeout == 0) {
+            timeout = TimeUnit.MINUTES.toMillis(queryTimeout);
```

Comment:
If  `queryTimeout` is already in milliseconds, I do not think you want this conversion here.